nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,311 @@
+        if (metadataKeyFilterInput != null && metadataKeyFilterInput.length() > 0) {
+            metadataKeyFilterRef.set(Pattern.compile(metadataKeyFilterInput));
+        } else {
+            metadataKeyFilterRef.set(null);
+        }
+
+        autoDetectParser = new AutoDetectParser();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ProcessorLog logger = this.getLogger();
+        final ObjectHolder<Map<String, String>> value = new ObjectHolder<>(null);
+        final Integer maxAttribCount = context.getProperty(MAX_NUMBER_OF_ATTRIBUTES).asInteger();
+        final Integer maxAttribLength = context.getProperty(MAX_ATTRIBUTE_LENGTH).asInteger();
+        final Integer contentBufferSize = context.getProperty(CONTENT_BUFFER_SIZE).evaluateAttributeExpressions().asInteger();
+        final String prefix = context.getProperty(METADATA_KEY_PREFIX).evaluateAttributeExpressions().getValue();
+
+        try {
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(InputStream in) throws IOException {
+                    try {
+                        Map<String, String> results = tika_parse(in, prefix, maxAttribCount, maxAttribLength, contentBufferSize);
+                        value.set(results);
+                    } catch (SAXException | TikaException e) {
+                        throw new IOException(e);
+                    }
+                }
+            });
+
+            // Write the results to attributes
+            Map<String, String> results = value.get();
+            if (results != null && !results.isEmpty()) {
+                flowFile = session.putAllAttributes(flowFile, results);
+            }
+
+            session.transfer(flowFile, SUCCESS);
+            session.getProvenanceReporter().modifyAttributes(flowFile, "media attributes extracted");
+        } catch (ProcessException e) {
+            logger.error("Failed to extract media metadata from {} due to {}", new Object[]{flowFile, e});
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, FAILURE);
+        }
+    }
+
+    private Map<String, String> tika_parse(InputStream sourceStream, String prefix, Integer maxAttribs, Integer maxAttribLen,
+                                           Integer contentBufSize) throws IOException, TikaException, SAXException {
+        final Metadata metadata = new Metadata();
+        final TikaInputStream tikaInputStream = TikaInputStream.get(sourceStream);
+
+        // as of June, 2016 the default BodyContentHandler used a 100,000 byte buffer
+        final BodyContentHandler bodyContentHandler = (contentBufSize != null) ? new BodyContentHandler(contentBufSize) : new BodyContentHandler();
```

Comment:
I'm actually wondering if we even want to use the BodyContentHandler here... the BodyContentHandler is used to get access to all of the extracted content, for example if we wanted to produce another flow file, or replace the current one with the extracted content, but it looks we aren't doing that here. There is a DefaultHandler with all no-ops that probably makes more sense for this scenario.